zkevm-circuits/src/circuit_tools/memory.rs

```diff
@@ -220,14 +286,11 @@ impl<F: Field, C: CellType> MemoryBank<F, C> {
         cb: &mut ConstraintBuilder<F, C>,
         is_first_row: Expression<F>,
     ) {
-        let lookups = cb.lookups.get(&self.table_tag()).unwrap();
-        let lookups = lookups
-            .iter()
-            .filter(|l| l.region_id == cb.region_id)
-            .collect::<Vec<_>>();
-        let condition = lookups
+        let condition = self
+            .table_conditions
```

Comment:
Why not get the condition from the lookups like before? This way we don't have to make most(?) memory functions mutable, the only mutable object is the constraint builder.

Reply:
Ok, i was thinking about the mutability as well and i feel like in general a better solution is to move the memory within CB, how's that?

Reply:
You mean a bit like how CellManager is part of the CB? I'm not sure it requires that, the current version reusing the lookups worked okay I think or do you see problems with that?

Reply:
Any update on this question?

Reply:
Will reuse cb's conditions :)